### ISSUE TITLE
Cleanup CFeeRate constructor (sat/vB vs BTC/kvB)

### ISSUE THREAD

**Issue by MarcoFalke** [bitcoin/bitcoin#23129]
Tracking issue for discussion in https://github.com/bitcoin/bitcoin/pull/20391#discussion_r668992659

**Comment by jonatack** [comment]
Since the merge of #21786 is the discussion still relevant?

**Comment by MarcoFalke** [comment]
Yes, I think so. That pull only fixed the AmountFromValue bug.

Cleaning up the constructor would be more of a refactor.

**Comment by jonatack** [comment]
Reminds me, @Xekyo and I prepared a refactoring a few months ago. @Xekyo, want to open it?

**Comment by willcl-ark** [comment]
@jonatack @murchandamus is this issue effectively solved via previous changes + #24305

**Comment by murchandamus** [comment]
@willcl-ark: I don’t think this is resolved. The state in `src/policy/feerate.cpp` still matches @maflcko’s comment from the linked PR that predates this issue by a day. I think #32750 is making some progress towards improvement, though.

**Comment by polespinasa** [comment]
Is this still relevant after #32750 is merged?

**Comment by maflcko** [comment]
@polespinasa I don't think this is resolved. To explain it a bit better, looking at the code:

https://github.com/bitcoin/bitcoin/blob/d41b503ae128ac36ef27e652d2935c6fe7981a79/src/wallet/rpc/spend.cpp#L223-L224

`AmountFromValue` is a function that returns a `CAmount` (i64 of satoshis). However, the `CFeeRate` constructor that is called takes a feerate per kvB:

https://github.com/bitcoin/bitcoin/blob/d41b503ae128ac36ef27e652d2935c6fe7981a79/src/policy/feerate.h#L44


So calling `CFeeRate{AmountFromValue("0.0000'0001")}` makes sense to call the BTC/kvB constructor.

However, calling `CFeeRate{AmountFromValue("0.001", 3)}` feels like a layer and type violation to parse sat/vB.

I guess if I am the only one bothered by it, the issue can be closed. The code behaves correctly and this is more in the style/refactor territory.

Maybe this can also be closed and transferred into https://github.com/bitcoin/bitcoin/issues/32093?

**Comment by maflcko** [comment]
Closing for now. Seems fine to just organically change this, the next time the code is touched.
